### PR TITLE
Initial reflected XSS unit tests

### PR DESCRIPTION
**Pull request by psiinon**
This is a reworking of #504 with the HTML defined in resource files.
The other PR has been left open to make it easier to compare the 2 approaches - the one we prefer (if any;) will be merged and the other closed.
Views?

### CODE REVIEW

**Comment by psiinon** [comment]
@thc202 @kingthorin any comments on this or #504 ?

File: test/org/zaproxy/zap/extension/ascanrules/ActiveScannerTest.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        respHeaderSB.append("Content-Type: text/html;charset=ISO-8859-1\r\n");
+        respHeaderSB.append("Content-Length: ");
+        respHeaderSB.append(msg.getResponseBody().length());
+        respHeaderSB.append("\r\n");
+        msg.setResponseHeader(respHeaderSB.toString());
+
+        return msg;
+    }
+    
+    public String getHtml (String name) {
```
Comment (kingthorin) [inline test/org/zaproxy/zap/extension/ascanrules/ActiveScannerTest.java]: All of the `getHtml` functions seem to have an extra space between func name and brackets.

**Comment by kingthorin** [comment]
Looks good to me (just that one minor issue with the spacing)

**Comment by psiinon** [comment]
Updated to fix those, and also replaced the config files with blank equivs and fixed the code to mock them correctly (thanks to @thc202 ;)

**Comment by kingthorin** [comment]
Looks good to me

**Comment by thc202** [comment]
Thanks!

**Pull request merged**